packages/account/imAccount/validators/ecdsaValidator.ts

```diff
@@ -0,0 +1,84 @@
+import * as ethers from "ethers"
+
+import ECDSAValidatorMetadata from "~packages/account/imAccount/abis/ECDSAValidator.json"
+import type { Validator } from "~packages/account/imAccount/validator"
+import type { HexString } from "~typing"
+
+export class ECDSAValidator implements Validator {
+  private node: ethers.JsonRpcProvider
+  private owner: ethers.Wallet
+  public contract: ethers.Contract
+
+  public constructor(opts: {
+    address: HexString
+    ownerPrivateKey: string
+    node: ethers.JsonRpcProvider
+  }) {
+    this.node = opts.node
+    this.owner = new ethers.Wallet(opts.ownerPrivateKey)
+    this.contract = new ethers.Contract(
+      opts.address,
+      ECDSAValidatorMetadata.abi,
+      this.node
+    )
+  }
+
+  public async sign(message: string | Uint8Array, metadata?: any) {
+    if (message.length < 66) {
+      message = ethers.encodeBytes32String(ethers.hexlify(message))
+    }
+    const digest = ethers.keccak256(
+      ethers.AbiCoder.defaultAbiCoder().encode(
+        ["bytes32", "address"],
+        [message, await this.contract.getAddress()]
+      )
+    )
+    const signingMsg = ethers.keccak256(
+      ethers.solidityPacked(
+        ["string", "string", "bytes"],
+        ["\x19Ethereum Signed Message:\n32", digest.length, digest]
```

Comment:
The `digest.length` is a `number`, not a `string` ([source](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/54b3f14346da01ba0d159114b399197fea8b7cda/contracts/utils/cryptography/ECDSA.sol#L184-L186)), and should be formatted as follows:
```
    const signingMsg = ethers.keccak256(
      ethers.solidityPacked(
        ["string", "string", "bytes"],
        [
          "\x19Ethereum Signed Message:\n",
          ethers.dataLength(digest).toString(),
          digest
        ]
      )
    )
```